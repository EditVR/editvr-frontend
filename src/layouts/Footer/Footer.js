/**
 * @file Footer.js
 * Exports EditVR's primary footer.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Typography, withStyles } from '@material-ui/core';

import FooterStyles from './Footer.style';
import footerLogo from '../../assets/4k-logo-reversed.svg';

const Footer = ({ classes }) => (
  <footer className={classes.wrapper}>
    <div className={classes.elements}>
      <a href="https://fourkitchens.com">
        <Typography type="body1" className={classes.text}>
          EditVR is a product of
        </Typography>
        <img src={footerLogo} alt="Four Kitchens" className={classes.logo} />
      </a>
    </div>
  </footer>
);

Footer.propTypes = {
  classes: PropTypes.shape({
    wrapper: PropTypes.string.isRequired,
    elements: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    logo: PropTypes.string.isRequired
  }).isRequired
};

export default withStyles(FooterStyles)(Footer);
