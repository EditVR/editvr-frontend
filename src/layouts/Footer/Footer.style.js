/**
 * @file Footer.style.js
 * Exports Footer component styles.
 */

export default () => ({
  wrapper: {
    marginTop: '5rem',
    padding: '0 1rem',
    position: 'relative',
    textAlign: 'center'
  },
  elements: {
    display: 'inline-block',
    position: 'relative'
  },
  logo: {
    maxWidth: '230px',
    width: '100%'
  },
  text: {
    color: 'rgba(255,255,255,.5)',
    marginBottom: '10px'
  },
  '@media (min-width: 380px)': {
    text: {
      marginLeft: '86px',
      marginBottom: '-20px'
    }
  }
});
