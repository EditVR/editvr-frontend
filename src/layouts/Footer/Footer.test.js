/**
 * @file Footer.test.js
 * Contains tests for Footer.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Footer from './Footer';

describe('<Footer />', () => {
  it('Matches its snapshot', () => {
    expect(renderer.create(<Footer />).toJSON()).toMatchSnapshot();
  });
});
