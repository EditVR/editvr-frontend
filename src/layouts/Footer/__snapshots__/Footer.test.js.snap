// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Matches its snapshot 1`] = `
<footer
  className="Footer-wrapper-1"
>
  <div
    className="Footer-elements-2"
  >
    <a
      href="https://fourkitchens.com"
    >
      <p
        className="MuiTypography-root-5 MuiTypography-body1-14 Footer-text-4"
        type="body1"
      >
        EditVR is a product of
      </p>
      <img
        alt="Four Kitchens"
        className="Footer-logo-3"
        src="4k-logo-reversed.svg"
      />
    </a>
  </div>
</footer>
`;
